gateway/src/apicast/policy/upstream/policy.lua

```diff
@@ -0,0 +1,65 @@
+--- Upstream policy
+-- This policy allows to modify the host of a request based on its path.
+
+local resty_resolver = require('resty.resolver')
+local resty_url = require('resty.url')
+local format = string.format
+local ipairs = ipairs
+local tonumber = tonumber
+local match = ngx.re.match
+local table = table
+
+local _M = require('apicast.policy').new('Upstream policy')
+
+local new = _M.new
+
+local function change_upstream(url)
+  ngx.ctx.upstream = resty_resolver:instance():get_servers(
```

Comment:
This code is very similar to what we already have in `Proxy.get_upstream()` https://github.com/3scale/apicast/blob/6049ab33581483b0e69fc1ff6ab2fea38aa4a696/gateway/src/apicast/proxy.lua#L180

If we think that there might be other policies that need the same functionality, we should refactor what we have there to avoid duplicating code in the policies.

Reply:
I'd rather address this on a separate PR as this is an internal change that will not affect the schema of the policy.

Reply:
Totally 👍